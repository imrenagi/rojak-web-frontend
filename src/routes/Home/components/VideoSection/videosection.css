.video-section {
  position: relative;
  text-align: center;
  margin-left: -15px;
  margin-right: -15px;

}

.title {
  font-size: 30px;
  padding-top: 20px;
  text-align: center;
}

.video .description {
  font-size: 24px;
  margin: 0px;
  text-align: center;
}

.video-thumbnail {
  padding: 50px;
}

.wrapper.style3 {
  border-bottom: 1px solid rgba(144, 144, 144, 0.25);
  background-color: #eff5f5;
}

.container-video {
  width: 60%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 15px;
  padding-right: 15px;
}
.embed-responsive {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);  
}

@media (max-width: 500px) .container-video {
  margin-left: 0;
  margin-right: 0;
}

@media (max-width: 568px) .container-video {
  margin-left: 40px;
  margin-right: 40px;
  max-width: 100%;
}

@media (max-width: 768px) .container-video {
  margin-left: 40px;
  margin-right: 40px;
  max-width: 100%;
}

@media (max-width: 992px) .container-video {
  max-width: 940px;
}

@media (max-width: 1200) .container-video {
  max-width: 1140px;
}

@media (min-width: 544px) .container-video {
  max-width: 576px;
}

@media (max-width: 380px) .container-video {
  margin: 0;
}