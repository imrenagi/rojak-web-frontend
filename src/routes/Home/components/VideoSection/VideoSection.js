import React from 'react'

import './videosection.css'

export default class VideoSection extends React.Component {
  render () {
    return (
      <div className='wrapper style3 video-section'>
        <div className='row title'>
          <h1>About Rojak</h1>
        </div>
        <div className='row description'>
          <p>Watch this video to understand more about Rojak</p>
        </div>
        <div className='video-thumbnail'>
          <div className='container-video'>
            <div className='embed-responsive embed-responsive-16by9'>
              <iframe
                className='embed-responsive-item'
                src='https://www.youtube.com/embed/3tmd-ClpJxA'
              />
            </div>
          </div>
        </div>
      </div>
    )
  }
}
