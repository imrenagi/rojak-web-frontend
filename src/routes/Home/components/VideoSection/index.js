import VideoSection from './VideoSection'

export default VideoSection
