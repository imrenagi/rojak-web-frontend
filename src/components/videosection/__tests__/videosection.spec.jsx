import React from 'react';

import renderer from 'react-test-renderer';

import { mount } from 'enzyme';

// I like to use chai but Jest uses jasmine expect statements
import chai from 'chai';

import VideoSection from '../VideoSection';

describe('Video Section', () => {
    describe('rendering', () => {
        it('should not render null', () => {
            // Assert
            // Mount the Component and check if it actually renders
            chai.expect(mount(<VideoSection />).html()).to.not.equal(null);
        });

        it('should render a wrapper div with className `video-section`', () => {
            // Arrange
            // Mount the component
            const wrapper = mount(<VideoSection />);
            // find the top level div with the correct className
            const wrapperDiv = wrapper
                .find('div')
                .filterWhere(div => div.prop('className').includes('wrapper'))
                .first(); // Get the first one since filterWhere returns a list

            // Assert
            chai.expect(wrapperDiv.exists()).to.be.true;
            chai.expect(wrapperDiv.prop('className').includes('video-section'))
                .to.be.true;
        });

        it('should render a title row', () => {
            // Arrange
            // Mount the component
            const wrapper = mount(<VideoSection />);

            const title = wrapper
                .find('div.row') // find all the div's with class row
                .filterWhere(div => div.prop('className').includes('title'))
                .first(); // Get the first one since filterWhere returns a list

            // Assert
            chai.expect(title.exists()).to.be.true;
            chai.expect(title.find('h1').exists()).to.be.true;
            chai.expect(title.find('h1').text()).to.equal('About Rojak');
        });

        it('should render a description', () => {
            // Arrange
            // Mount the component
            const wrapper = mount(<VideoSection />);

            const description = wrapper
                .find('div.row') // find all the div's with class row
                .filterWhere(div =>
                    div.prop('className').includes('description')
                )
                .first(); // Get the first one since filterWhere returns a list

            // Assert
            chai.expect(description.exists()).to.be.true;
            chai.expect(description.find('p').exists()).to.be.true;
            chai
                .expect(description.find('p').text())
                .to.equal('Watch this video to understand more about Rojak');
        });

        it('should render a video in an iframe', () => {
            // Arrange
            // Mount the component
            const wrapper = mount(<VideoSection />);

            const video = wrapper.find('div.video-thumbnail');

            // Assert
            chai.expect(video.exists()).to.be.true;
            chai.expect(video.find('div.container-video').exists()).to.be.true;
            chai.expect(video.find('div.embed-responsive').exists()).to.be.true;
            chai.expect(video.find('iframe').exists()).to.be.true;
            chai
                .expect(video.find('iframe').prop('src'))
                .to.equal('https://www.youtube.com/embed/3tmd-ClpJxA');
        });
    });

    describe('snapshots', () => {
        it('should match the snapshot', () => {
            // Assert
            expect(
                renderer.create(<VideoSection />).toJSON()
            ).toMatchSnapshot();
        });
    });
});
