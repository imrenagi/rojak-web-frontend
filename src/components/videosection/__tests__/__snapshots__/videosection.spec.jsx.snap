// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Section snapshots should match the snapshot 1`] = `
<div
  className="wrapper style3 video-section"
>
  <div
    className="row title"
  >
    <h1>
      About Rojak
    </h1>
  </div>
  <div
    className="row description"
  >
    <p>
      Watch this video to understand more about Rojak
    </p>
  </div>
  <div
    className="video-thumbnail"
  >
    <div
      className="container-video"
    >
      <div
        className="embed-responsive embed-responsive-16by9"
      >
        <iframe
          className="embed-responsive-item"
          src="https://www.youtube.com/embed/3tmd-ClpJxA"
        />
      </div>
    </div>
  </div>
</div>
`;
